src/software/ai/navigator/path_planner/enlsvg_path_planner.cpp

```diff
@@ -0,0 +1,199 @@
+#include "enlsvg_path_planner.h"
+
+EnlsvgPathPlanner::EnlsvgPathPlanner(const Rectangle &navigable_area,
+                                     const std::vector<ObstaclePtr> &obstacles,
+                                     double grid_boundary_offset, double resolution)
+    : resolution(resolution),
+      num_grid_rows(static_cast<int>(round(navigable_area.xLength() / resolution))),
+      num_grid_cols(static_cast<int>(round(navigable_area.yLength() / resolution))),
+      origin(navigable_area.negXNegYCorner()),
+      max_navigable_y_enlsvg_point(
+          convertPointToEnlsvgPoint(navigable_area.posXPosYCorner()).y),
+      max_navigable_x_enlsvg_point(
+          convertPointToEnlsvgPoint(navigable_area.posXPosYCorner()).x),
+      grid(std::make_unique<EnlsvgGrid>(num_grid_rows, num_grid_cols))
+{
+    createObstaclesInGrid(obstacles, grid_boundary_offset);
+    algo = std::make_unique<const EnlsvgAlgorithm>(*grid);
+    mem  = std::make_unique<EnlsvgMemory>(*algo);
+}
+
+void EnlsvgPathPlanner::createObstaclesInGrid(const std::vector<ObstaclePtr> &obstacles,
+                                              double boundary_margin) const
+{
+    double offset_in_enlsvg = boundary_margin / resolution;
+    for (int x = 0; x < offset_in_enlsvg; ++x)
+    {
+        for (int y = 0; y < num_grid_cols; ++y)
+        {
+            grid->setBlocked(x, y, true);
+            grid->setBlocked(num_grid_rows - 1 - x, y, true);
+        }
+    }
+    for (int y = 0; y < offset_in_enlsvg; ++y)
+    {
+        for (int x = 0; x < num_grid_rows; ++x)
+        {
+            grid->setBlocked(x, y, true);
+            grid->setBlocked(x, num_grid_cols - 1 - y, true);
+        }
+    }
+
+    for (auto &obstacle : obstacles)
+    {
+        auto blocked_points = obstacle->rasterize(resolution);
+
+        for (Point &blocked_point : blocked_points)
+        {
+            EnlsvgPoint blocked_coord = convertPointToEnlsvgPoint(blocked_point);
+            if (isCoordNavigable(blocked_coord))
+            {
+                grid->setBlocked(blocked_coord.x, blocked_coord.y, true);
+            }
+        }
+    }
+}
+
+bool EnlsvgPathPlanner::isCoordNavigable(const EnlsvgPoint &ep) const
+{
+    return (ep.x >= 0 && ep.x < max_navigable_x_enlsvg_point) &&
+           (ep.y >= 0 && ep.y < max_navigable_y_enlsvg_point);
+}
+
+std::optional<Path> EnlsvgPathPlanner::findPath(const Point &start, const Point &end,
+                                                const Rectangle &navigable_area,
+                                                const std::vector<ObstaclePtr> &ignored)
```

Comment:
you can actually not give it a name
```suggestion
                                                const std::vector<ObstaclePtr> &)
```

Reply:
fixed